src/Storage/Database/Schema/Types/JsonType.php

```diff
@@ -0,0 +1,127 @@
+ *
+ * @author Baptiste Clavié <clavie.b@gmail.com>
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+final class JsonType extends Types\Type
+{
+    const JSON = 'json';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
+    {
+        return $platform->getJsonTypeDeclarationSQL($fieldDeclaration);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        try {
+            $encoded = Json::dump($value);
+        } catch (DumpException $e) {
+            throw Types\ConversionException::conversionFailed($value, 'json');
+        }
+
+        return $encoded;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform)
+    {
+        if ($value === null || $value === '') {
+            return null;
+        }
+
+        if (is_resource($value)) {
+            $value = stream_get_contents($value);
+        }
+
+        try {
+            $val = Json::parse($value, true);
+        } catch (ParseException $e) {
+            throw Types\ConversionException::conversionFailed($value, $this->getName());
+        }
+
+        return $val;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'json';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresSQLCommentHint(AbstractPlatform $platform)
+    {
+        return !$platform->hasNativeJsonType();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMappedDatabaseTypes(AbstractPlatform $platform)
+    {
+        $type = 'json';
+        $platformName = $platform->getName();
+
+        if ($platformName === 'sqlite') {
+            /** @see \Doctrine\DBAL\Platforms\SqlitePlatform::getClobTypeDeclarationSQL */
+            $type = 'clob';
+        } elseif ($platformName === 'mysql') {
+            if ($platform instanceof Platforms\MySQL57Platform) {
+                /** @see \Doctrine\DBAL\Platforms\MySQL57Platform::getJsonTypeDeclarationSQL */
+                $type = 'json';
+            } else {
+                /** @see \Doctrine\DBAL\Platforms\MySqlPlatform::getClobTypeDeclarationSQL */
+                $type = 'longtext';
+            }
+        } elseif ($platformName === 'postgresql') {
+            if ($platform instanceof Platforms\PostgreSQL92Platform) {
+                /** @see \Doctrine\DBAL\Platforms\PostgreSQL92Platform::getJsonTypeDeclarationSQL */
+                $type = 'json';
+            } else {
+                /** @see \Doctrine\DBAL\Platforms\PostgreSqlPlatform::getClobTypeDeclarationSQL */
+                $type = 'text';
+            }
+        }
+
+        $platform->registerDoctrineTypeMapping($type, static::JSON);
```

Comment:
You don't need to call this. Doctrine does it for you [here](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Platforms/AbstractPlatform.php#L247-L249).

Reply:
Then something weird is happening, I originally dropped it but things went :boom: 

I'll dig now.

Reply:
I don't know what you're talking about, that line isn't really there … this is your imagination :stuck_out_tongue_winking_eye: 

Oh look … :smile_cat: